atomic_reactor/plugins/post_fetch_worker_metadata.py

```diff
@@ -0,0 +1,73 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+from atomic_reactor.plugin import PostBuildPlugin
+from atomic_reactor.plugin import PluginFailedException
+from atomic_reactor.plugins.build_orchestrate_build import get_worker_build_info
+from atomic_reactor.constants import PLUGIN_FETCH_WORKER_METADATA_KEY
+
+
+class FetchWorkerMetadataPlugin(PostBuildPlugin):
+    """
+    Fetch worker metadata from each platform and return a dict of
+    each platform's metadata.
+
+    """
+
+    key = PLUGIN_FETCH_WORKER_METADATA_KEY
+
+    def __init__(self, tasker, workflow):
+        """
+        constructor
+
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        """
+        super(FetchWorkerMetadataPlugin, self).__init__(tasker, workflow)
+
+    def run(self):
+        """
+        Run the plugin.
+        """
+
+        metadatas = {}
+
+        # get all the build annotations and labels from the orchestrator
+        build_result = self.workflow.build_result
+
+        if not build_result:
+            raise PluginFailedException('No build_result for FetchWorkerMetadataPlugin')
+
+        annotations = build_result.annotations
+        worker_builds = annotations['worker-builds']
+
+        for platform in worker_builds.keys():
+            # retrieve all the workspace data
+            build_info = get_worker_build_info(self.workflow, platform)
+            osbs = build_info.osbs
+
+            build_annotations = worker_builds[platform]
+
+            kind = "configmap/"
+            cmlen = len(kind)
+            cm_key_tmp = build_annotations['metadata_fragment']
+            cm_frag_key = build_annotations['metadata_fragment_key']
+
+            if not cm_key_tmp or not cm_frag_key or cm_key_tmp[0:cmlen] != kind:
+                self.log.warning("Bad ConfigMap annotations for platform %s" % platform)
```

Comment:
For logging calls it's good practice to pass arguments for %-formatting as separate parameters:
`self.log.warning("Bad ConfigMap ... %s", platform)`